src/System.Collections/tests/Generic/Dictionary/Dictionary.Generic.Tests.cs

```diff
@@ -154,5 +154,75 @@ public void IReadOnlyDictionary_Generic_Values_ContainsAllCorrectValues(int coun
         }
 
         #endregion
+
+        #region Remove(TKey)
+
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void Dictionary_Generic_RemoveKey_ValidKeyNotContainedInDictionary(int count)
+        {
+            Dictionary<TKey, TValue> dictionary = (Dictionary<TKey, TValue>)GenericIDictionaryFactory(count);
+            TValue value;
+            TKey missingKey = GetNewKey(dictionary);
+
+            Assert.False(dictionary.Remove(missingKey, out value));
+            Assert.Equal(count, dictionary.Count);
+            Assert.Equal(default(TValue), value);
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void Dictionary_Generic_RemoveKey_ValidKeyContainedInDictionary(int count)
+        {
+            Dictionary<TKey, TValue> dictionary = (Dictionary<TKey, TValue>)GenericIDictionaryFactory(count);
+            TKey missingKey = GetNewKey(dictionary);
+            TValue value;
+
+            dictionary.Add(missingKey, CreateTValue(34251));
+            Assert.True(dictionary.Remove(missingKey, out value));
+            Assert.Equal(count, dictionary.Count);
+            Assert.Equal(CreateTValue(34251), value);
```

Comment:
- [x] Used CreateTValue once.
- [x] Actually this was to create and check the value consistency... Changed random seed to `count` that is passed to the test method.
- [x] Added TryGetValue